src/pathways/escher.service.ts

```diff
@@ -1,100 +1,135 @@
+            this.map.new_reaction_from_scratch(
+                firstReaction.id,
+                startCoordinates,
+                self.direction
+            );
+            _.forEach(restReactionWithNodes, ([reaction, nodeName]) => {
+                // This would be nicer with _.maybe
+                // Note: Nodes behaves like an array, but it's a dicitionary
+                // with sequential integer indices
+                const nodeIndex = _.findKey(this.map.nodes, {name: nodeName});
+                if (!_.isUndefined(nodeIndex)) {
+                    this.map.new_reaction_for_metabolite(
+                        reaction.id,
+                        nodeIndex,
+                        self.direction
+                    );
+                }
+            });
+
+            // Set metabolites node to primary if it's not among the metabolites.
+            this.map.nodes = _.map(this.map.nodes, (node) => {
+                return _.assign(node, {
+                    node_is_primary: _.includes(self.metabolitesNames, node.name),
+                })
+            });
+
+            this.map.draw_everything();
+            this.map.zoom_extent_canvas();
+            this.map.select_none();
+        }
+    }
 
-	constructor() {
-		this.direction = -90;
-		this.metabolitesNames = [];
-		this.reactions = [];
-		this.self = this;
-		this.options = {
-			// just show the zoom buttons
-			menu: 'zoom',
-			// do not use the smooth pan and zoom option
-			use_3d_transform: false,
-			// no editing in this map
-			enable_editing: true,
-			// show the descriptive names
-			identifiers_on_map: 'name',
-			// hide secondary metabolites
-			hide_secondary_metabolites: true,
-			// don't ask before quiting
-			never_ask_before_quit: true,
-			// disable keyboard shortcuts
-			enable_keys: false,
-			first_load_callback: () => {
-				let currentThis = eval('this');
-				this.startCoordinates = {x: 0, y: (this.reactions.length/2)*400 };
-				for (let i = this.reactions.length - 1; i >= 0; i -= 1) {
-					if (i === this.reactions.length - 1) {
-						currentThis.map.new_reaction_from_scratch(
-							this.reactions[i].id,
-							this.startCoordinates,
-							this.direction
-						);
-					} else {
-						for (let k in currentThis.map.nodes) {
-							if (currentThis.map.nodes[k].name === this.metabolitesNames[i]) {
-								currentThis.map.new_reaction_for_metabolite(
-									this.reactions[i].id,
-									k,
-									this.direction);
-								break;
-							}
-						}
-					}
-				}
-				for (let n in currentThis.map.nodes) {
-					currentThis.map.nodes[n].node_is_primary =
-						(this.metabolitesNames.indexOf(currentThis.map.nodes[n].name) != -1);
-				}
-				currentThis.map.draw_everything();
-				currentThis.map.zoom_extent_canvas();
-				currentThis.map.select_none();
-			}
-		}
-	};
+    /** Flips the reaction in the other direction by changing
```

Comment:
Put the text into a new line. Also, you should add documentation blocks for the other functions, which are currently missing them.